cli/azd/pkg/commands/pipeline/pipeline_manager.go

```diff
@@ -420,3 +446,42 @@ func (manager *PipelineManager) Configure(ctx context.Context) (
 		PipelineLink:   strings.TrimSuffix(ciPipeline.remote, ".git"),
 	}, nil
 }
+
+func checkRoleAssignments(
+	ctx context.Context,
+	azcli azcli.AzCli,
+	subscriptionId string,
+	principalId string,
+	console input.Console,
+) error {
+	// Find the specified role in the subscription scope
+	scope := azure.SubscriptionRID(subscriptionId)
+	roles, err := azcli.GetUserRoleDefinitionName(ctx, subscriptionId, scope, principalId)
+	if err != nil {
+		return err
+	}
+	for _, name := range roles {
+		if name == "Owner" || name == "User Access Administrator" {
+			return nil
+		}
+	}
+
+	message := fmt.Sprintf(
```

Comment:
I think we can improve the message here a little.  The built in roles most likely exist, it's just that the current user who is running `pipeline config` is not assigned them.

Perhaps something like:

"warning: Your user account does not appear to have either the Owner or User Access Administrator role assigned to it.  This operation will attempt to grant permissions to the newly created service principal and this will  fail unless you have been granted these permissions with a custom role assignment.  Do you want to continue with configuring your pipeline?"